---
string fringe_coding_mode